### PR TITLE
45730 UI test foundation

### PR DESCRIPTION
**Pull request by jigar15426**
UI test foundation setup

### CODE REVIEW

File: Tests/playwright.config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"use strict";
```
Comment (nevillejrbrown) [inline Tests/playwright.config.js:1]: Does this file have to be checked in?  Isn't it generated in the build process?

File: Tests/playwright.config.js.map [hunk 2]
```diff
@@ -0,0 +1 @@
...M,GAAyB;IACnC,OAAO,EAAE,aAAa;IACvB,0CAA0C;IAEzC,wCAAwC;IACxC,OAAO,EAAE,EAAE,GAAG,IAAI;IAClB,MAAM,EAAE;QACN;;;WAGG;QACH,OAAO,EAAE,KAAK;KACf;IACD,iFAAiF;IACjF,UAAU,EAAE,CAAC,CAAC,OAAO,CAAC,GAAG,CAAC,EAAE;IAC5B,sBAAsB;IACtB,OAAO,EAAE,OAAO,CAAC,GAAG,CAAC,EAAE,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;IAC/B,sCAAsC;IACtC,OAAO,EAAE,OAAO,CAAC,GAAG,CAAC,EAAE,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,SAAS;IACvC,qEAAqE;IACrE,QAAQ,EAAE,MAAM;IAChB,wGAAwG;IACxG,GAAG,EAAE;QACH,oFAAoF;QACpF,aAAa,EAAE,CAAC;QAEhB,6DAA6D;QAC7D,oCAAoC;QAEpC,+FAA+F;QAC/F,KAAK,EAAE,gBAAgB;KACxB;IACD,2CAA2C;IAC3C,QAAQ,EAAE;QACR;YACE,IAAI,EAAE,UAAU;YAChB,GAAG,eACE,cAAO,CAAC,gBAAgB,CAAC,CAE7B;SACF;QAED,KAAK;QACL,sBAAsB;QACtB,YAAY;QACZ,sCAAsC;QACtC,QAAQ;QACR,MAAM;QAEN,IAAI;QACJ,oBAAoB;QACpB,WAAW;QACX,oCAAoC;QACpC,OAAO;QACP,KAAK;QAEL,oCAAoC;QACpC,IAAI;QACJ,2BAA2B;QAC3B,WAAW;QACX,6BAA6B;QAC7B,OAAO;QACP,KAAK;QACL,IAAI;QACJ,2BAA2B;QAC3B,WAAW;QACX,+BAA+B;QAC/B,OAAO;QACP,KAAK;QAEL,oCAAoC;QACpC,IAAI;QACJ,4BAA4B;QAC5B,WAAW;QACX,yBAAyB;QACzB,OAAO;QACP,KAAK;QACL,IAAI;QACJ,2BAA2B;QAC3B,WAAW;QACX,yBAAyB;QACzB,OAAO;QACP,KAAK;KACN;IAED,yEAAyE;IACzE,8BAA8B;IAE9B,yDAAyD;IACzD,eAAe;IACf,8BAA8B;IAC9B,gBAAgB;IAChB,KAAK;CACN,CAAC;AAEF,kBAAe,MAAM,CAAC"}
```
Comment (nevillejrbrown) [inline Tests/playwright.config.js.map:1]: Does this file have to be checked in? Isn't it generated in the build process?

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { test, expect, chromium, Page, Browser, BrowserContext } from '@playwright/test';
+import * as data from "../../configuration/pageObject.json";
+import * as app from "../../Configuration/appConfig.json";
+import headerPage from '../../pageObject/home.page';
+import { injectAxe, getViolations, checkA11y} from 'axe-playwright'
+
+test.describe("Goto maritime-safety-information Home Page ", ()=> {
+  let browser: Browser;
```
Comment (nevillejrbrown) [inline Tests/AutoTests/AccessebilityTest/accessebility.spec.ts:8]: You don't need to re-define Browser, BrowserContext & Page variables as these are already created by the framework and can be passed into the tests as parameters as per our template project https://github.com/UKHO/playwright-template/blob/main/tests/a11y/a11y.spec.ts

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  let context: BrowserContext;
+  let page: Page;
+  let homepage:headerPage;
+  test.beforeAll(async () => {
+    browser = await chromium.launch();
```
Comment (nevillejrbrown) [inline Tests/AutoTests/AccessebilityTest/accessebility.spec.ts:13]: You don't need to launch the browser explicitly here as this is just overriding automatic functionality of the framework

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  test.afterAll(async () => {
+  
+     await browser.close();
```
Comment (nevillejrbrown) [inline Tests/AutoTests/AccessebilityTest/accessebility.spec.ts:18]: You don't need to close the browser explicitly here as this is just overriding automatic functionality of the framework

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  });
+  
+  test.afterEach(async () => {
```
Comment (nevillejrbrown) [inline Tests/AutoTests/AccessebilityTest/accessebility.spec.ts:21]: You don't need to do these things explicitly here as this is just overriding automatic functionality of the framework

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    await page.close();
+    await context.close();
+  });
+  test.beforeEach(async () => {
```
Comment (nevillejrbrown) [inline Tests/AutoTests/AccessebilityTest/accessebility.spec.ts:25]: you don't need to create the page & context here; they are already accessible as parameters as per examples on https://github.com/UKHO/playwright-template/blob/main/tests/a11y/a11y.spec.ts

File: Tests/Configuration/pageObject.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (nevillejrbrown) [inline Tests/Configuration/pageObject.json:1]: Could you please use separate page object models for each page.  We have created some example for you in this template project: https://github.com/UKHO/playwright-template

A good example of a page object model is https://github.com/UKHO/playwright-template/blob/main/tests/page-object-model/formpage.ts

**Review by nevillejrbrown — CHANGES_REQUESTED** [review]
Can you please take a look at the UKHO template project https://github.com/UKHO/playwright-template and follow that?  Our apologies not to show you this earlier, but we've only just created it and it's still in review.

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { test, expect, chromium, Page, Browser, BrowserContext } from '@playwright/test';
+import * as data from "../../configuration/pageObject.json";
+import * as app from "../../Configuration/appConfig.json";
+import headerPage from '../../pageObject/home.page';
+import { injectAxe, getViolations, checkA11y} from 'axe-playwright'
+
+test.describe("Goto maritime-safety-information Home Page ", ()=> {
```
[the remark this answers is not in the document]
Comment (jigar15426) [reply Tests/AutoTests/AccessebilityTest/accessebility.spec.ts]: resolved

File: Tests/playwright.config.js.map (hunk 2 above)
Comment (jigar15426) [reply Tests/playwright.config.js.map]: resolved

File: Tests/playwright.config.js (hunk 1 above)
Comment (jigar15426) [reply Tests/playwright.config.js]: resolved

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts (hunk 3 above)
Comment (jigar15426) [reply Tests/AutoTests/AccessebilityTest/accessebility.spec.ts]: resolved for the browser

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts (hunk 5 above)
Comment (jigar15426) [reply Tests/AutoTests/AccessebilityTest/accessebility.spec.ts]: resolved

File: Tests/AutoTests/AccessebilityTest/accessebility.spec.ts (hunk 7 above)
Comment (jigar15426) [reply Tests/AutoTests/AccessebilityTest/accessebility.spec.ts]: resolved and accessible as parameters

File: Tests/Configuration/pageObject.json (hunk 8 above)
Comment (jigar15426) [reply Tests/Configuration/pageObject.json]: resolved and remove the pageobject json file, already we are following page object model for the individual pages

File: Tests/AutoTests/FunctionalTest/functional.spec.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import { test, expect, chromium, Page, Browser, BrowserContext } from '@playwright/test';
+import * as app from "../../Configuration/appConfig.json";
+import home from '../../pageObject/home.page';
+
+test.describe("Goto maritime-safety-information Home Page", ()=> {
```
Comment (nevillejrbrown) [inline Tests/AutoTests/FunctionalTest/functional.spec.ts:5]: Can you please change this test template to be like accessebility.spec.ts ?

**Review by nevillejrbrown — APPROVED** [review]
Please can you action my comment regarding Tests/AutoTests/FunctionalTest/functional.spec.ts?  Otherwise approved.

File: Tests/AutoTests/FunctionalTest/functional.spec.ts (hunk 10 above)
Comment (jigar15426) [reply Tests/AutoTests/FunctionalTest/functional.spec.ts:4]: the functional spec file template is resolved as accessebility spec file

**Review by kajalkundu — APPROVED** [review]
Looks good to me.

**Pull request merged**